### ISSUE TITLE
Outputting / serializing to the Nickel format

### ISSUE THREAD

**Issue by rob-p** [tweag/nickel#1616]
We are considering replacing jsonnet with nickel in one of our tools.  However, a major use case for us is the ability to "partially instantiate" a template.  That is, we'd like to be able to evaluate a configuration file, output the result to a configuration file *in the same format* and then potentially modify and re-evaluate that.

The nickel `convert` command can output to several formats, but can't write its own results back to a nickel file.  Is there a simple way to do this that I'm currently overlooking?  If not, how difficult might this be (I assume not too bad since once you parse and evaluate, we have the whole AST)?  Any suggestions would be appreciated! (cc @DongzeHe).

**Comment by yannham** [comment]
Hi, glad to hear that!

Currently the UI is not ideal: we would like to have an `eval` subcommand, but it's just the default behavior - that is, if you just do `nickel -f somefile.ncl`, then it will evaluate the file and pretty-print the result as valid Nickel code.

Note that the pretty-printer is tested, but the output code could have some differences with the original code, because of initial code transformations or desugaring (that is, some stuff might be evaluated away). There's also something fishy about contracts application being duplicated a lot that I'm investigating right now, but it's unrelated to pretty-printing - evaluation seems to be at fault here.

In conclusion, the default command of `nickel` should be capable of doing what you want, but it hasn't been used that way yet, so you may unveil some quirk. Happy to help if that happens, though

**Comment by rob-p** [comment]
Thankd for the quick response! This is very useful. We plan to use nickel as a library rather than a cmdline tool, so we van just eval a config and then output it again :). I'll open a different issue is we run into any specific problems. Thanks!
